src/ProjectConfiguration.ts

```diff
@@ -57,55 +58,71 @@ export const LOCALES = Object.freeze({
   'zh-tw': 'Chinese (T)',
 });
 
-const permissionTypes = Object.freeze({
-  access_activity: {
+const permissionTypes = [
+  {
+    key: 'access_activity',
     name: 'Activity',
     // tslint:disable-next-line:max-line-length
     description: 'Read user activities for today (distance, calories, steps, elevation and active minutes), and daily goals.',
   },
-  access_user_profile: {
+  {
+    key: 'access_user_profile',
     name: 'User Profile',
     // tslint:disable-next-line:max-line-length
     description: 'Read non-identifiable personal information (gender, age, height, weight, resting HR, basal metabolic rate, stride, HR zones).',
   },
-  access_heart_rate: {
+  {
+    key: 'access_heart_rate',
     name: 'Heart Rate',
     description: 'Application may read the heart-rate sensor in real-time.',
   },
-  access_location: {
+  {
+    key: 'access_location',
     name: 'Location',
     description: 'Application and companion may use GPS.',
   },
-  access_internet: {
+  {
+    key: 'access_internet',
     name: 'Internet',
     description: 'Companion may communicate with the Internet using your phone data connection.',
   },
-  run_background: {
+  {
+    key: 'run_background',
     name: 'Run in background',
     description: 'Companion may run even when the application is not actively in use.',
   },
-});
+  {
+    key: 'access_exercise',
+    name: 'Exercise Tracking',
+    description: 'Application may track an exercise.',
+    sdkMajorVersion: 3,
+  },
+];
 
-const restrictedPermissionTypes = Object.freeze({
-  fitbit_token: {
+const restrictedPermissionTypes = [
+  {
+    key: 'fitbit_token',
     name: '[Restricted] Fitbit Token',
     description: 'Access Fitbit API token.',
   },
-  external_app_communication: {
+  {
+    key: 'external_app_communication',
     name: '[Restricted] External Application Communication',
     description: 'Allows communication between external mobile applications and companion.',
   },
-  access_secure_exchange: {
+  {
+    key: 'access_secure_exchange',
     name: '[Restricted] Secure Exchange',
     description: 'Allows securing any data and verifying that data was secured',
   },
-});
-
-const allPermissionTypes = { ...permissionTypes, ...restrictedPermissionTypes };
-
-export function getPermissionTypes({ includeRestrictedPermissions = false }) {
-  if (includeRestrictedPermissions) return allPermissionTypes;
-  return permissionTypes;
+];
+
+function getAllPermissionTypes() {
+  return [
+    ...restrictedPermissionTypes,
+    ...permissionTypes.filter(permission =>
+      !permission.sdkMajorVersion || sdkVersion().major >= permission.sdkMajorVersion),
```

Comment:
This is a perfect opportunity to use semver ranges.
```ts
{
  key: 'access_exercise',
  sdkVersion: '>=3.0.0',
}
```
```suggestion
      !permission.sdkVersion || semver.satisfies(sdkVersion(), permission.sdkVersion),
```